src/sonic-eventd/rsyslog_plugin/rsyslog_plugin.cpp

```diff
@@ -2,20 +2,28 @@
 #include <vector>
 #include <fstream>
 #include <regex>
+#include <ctime>
+#include <unordered_map>
 #include "rsyslog_plugin.h"
 #include "json.hpp"
 
 using json = nlohmann::json;
 
-bool RsyslogPlugin::onMessage(string msg) {
+bool RsyslogPlugin::onMessage(string msg, lua_State* luaState) {
     string tag;
     event_params_t paramDict;
-    if(!m_parser->parseMessage(msg, tag, paramDict)) {
+    if(!m_parser->parseMessage(msg, tag, paramDict, luaState)) {
         SWSS_LOG_DEBUG("%s was not able to be parsed into a structured event\n", msg.c_str());
         return false;
     } else {
+	string timestamp = paramDict["timestamp"];
+        string formattedTimestamp = m_timestampFormatter->changeTimestampFormat(paramDict["timestamp"]);
+	if(timestamp.empty()) {
```

Comment:
Do you intend to check formattedTimestamp ?
On failure, don't send timestamp. 
The publisher API will add one.

Reply:
Done